app/models/journaled/bulk_delivery.rb

```diff
@@ -0,0 +1,45 @@
+class Journaled::BulkDelivery
+  def initialize(records:, app_name:)
```

Comment:
My last commit refactored into this `records` which has the shape
```
[ [DATA, KEY], [DATA, KEY], ...]
```

This couples the data and key together, which I like. But its also a bit opaque I feel like. I _could_ make the inner arrays into hashes, but this is going to be serialized as yml and I don't really want to have to serialize the hash keys in a big bulk list.

Previously I had two params `serialized_events` and `partition_keys` which were `zip`ed together and iterated over. This required both arrays to have the same/matching order.

I _think_ I like the `records` abstraction more, but wanted thoughts!

Reply:
I think array of arrays is totally functional here since we can guarantee the order of elements in the inner array